hack/compare-apt-repo-versions.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -o errexit
+
+SCRIPT_NAME="${0##*/}"
+readonly SCRIPT_NAME
+
+SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
+readonly SCRIPT_DIR
+
+usage() {
+    echo "Script to compare the apt repo version of two Garden Linux releases."
+    echo "Usage: $SCRIPT_NAME VERSION_A VERSION_B"
+    echo "Example: $SCRIPT_NAME 1443.10 1587.0"
+    exit 1
+}
+
+main() {
+    [[ $# -ge 2 ]] || usage
+    [[ -n "$1" ]] || usage
+    [[ -n "$2" ]] || usage
+    local VERSION_A="${1}"; shift
+    local VERSION_B="${1}"; shift
+
+    TEMP_DIR=$(mktemp -d)
+
+    trap 'rm -rf $TEMP_DIR' EXIT
+
+    curl -s https://packages.gardenlinux.io/gardenlinux/dists/"$VERSION_A"/main/binary-amd64/Packages.gz | gunzip > "$TEMP_DIR"/"$VERSION_A"
+    curl -s https://packages.gardenlinux.io/gardenlinux/dists/"$VERSION_B"/main/binary-amd64/Packages.gz | gunzip > "$TEMP_DIR"/"$VERSION_B"
+
+    python3 "$SCRIPT_DIR"/parse-aptsource.py "$TEMP_DIR"/"$VERSION_A" > "$TEMP_DIR"/a
+    python3 "$SCRIPT_DIR"/parse-aptsource.py "$TEMP_DIR"/"$VERSION_B" > "$TEMP_DIR"/b
+
+    diff "$TEMP_DIR"/a "$TEMP_DIR"/b
```

Comment:
```suggestion
    diff -U 0 --minimal  "$TEMP_DIR"/a "$TEMP_DIR"/b | grep -v '^@'
```

I suggest a different output format that is a bit easier to read and parse.

Reply:
` -U 0`:  removes the context lines that did not change and uses the `+ -` format 

`--minimal`:  not sure if we need it, but it sounds cool 
> -d, --minimal
              try hard to find a smaller set of changes

` | grep -v '^@'`: removes lines starting with @, which are part of output from diff to indicate where a "chunk" starts

Reply:
We could further improve usability by adding more flags to this tool.

`--ignore-removed`: does a `| grep -v '^-'` to remove lines that show what packages were 
`--only-new`: only shows packages that did not exist before 
`--only-version-changes`: only shows packages that existed before in a different version

just food for thought. Not an issue.

Reply:
Thanks for your suggestions. Another option I was thinking about is `side-by-side`.. does not work with `U 0` but might also be useful.

<img width="897" alt="Screenshot 2024-08-06 at 14 14 26" src="https://github.com/user-attachments/assets/3e544173-64a2-4cbc-8b9e-4fe2dc8385af">

I'd like to merge it with your suggestion and then evolve the tool as we see fit. I'd like to avoid too complex argument parsing, but some more convenience features would be welcome for sure.